src/applications/common/schemaform/widgets/SelectWidget.jsx

```diff
@@ -15,6 +15,38 @@ function getValue(event) {
   return event.target.value;
 }
 
+function getOptions(enumOptions, labels) {
```

Comment:
The implementation seems specific to our use-case. Should we support optgroups in the common widget at all? If so, is this the right abstraction for doing so? These aren't rhetorical, I'm genuinely curious 😛

Reply:
Good question, @markgreenburg, @goldenmeanie do we know whether we want to use a select with groups for the UI here? multiple dropdowns? an autosuggest field?
If there is interest in using option groups in a dropdown/select, whether that belongs in the common widget is a good question, perhaps @jbalboni can shed some light on that?

Reply:
Another question: just because the EVSS spreadsheet categorizes these document types, do we have to? The categories don't really seem to be that helpful. If we just didn't categorize these issues at all (knowing that @goldenmeanie probably has some UX alternatives to using `select` he could cook up post-MVP), that would mean we wouldn't need common forms lib changes, and could strip out a ton of complexity. While the list is long, I don't know that we're helping the veteran by categorizing it. Thoughts? @cvalarida @cehsu

Reply:
Good question @markgreenburg, whether to use categories at all seems like a @peggygannon @goldenmeanie question.

Reply:
I think making this an autosuggest field would be more useful that the categorization, but if we want to make optgroups a pattern we use on forms, we need to figure out a generic approach that will work for everyone.

Reply:
Seems like we should strip out the optgroup / common changes and just go with an uncategorized list for now (we can change to using autosuggest later, if needed). Might be worth opening up a separate issue for discovery on adding optgroups to common forms code, if that's something we want to pursue (which is something we could decide as part of discovery).

Reply:
The groups are pretty exciting to me. This is big list for something less regular than states or other familiar groups. We could potentially do a conditional radio cluster instead, if the infrastructure to support `optgroup` seems fraught.

Reply:
Thanks @goldenmeanie, that makes sense. I think what I'm trying to get at is that it'd be easier to separate the implementation of the optgroups / whatever the final UX will be from this PR. It seems to me that this PR's focus is on making sure that we're able to attach the right document type (id?) to each upload. Support for option groups in common forms lib seems tangential to what we're trying to accomplish here, and might require more thought given that we'd want to settle on an approach that works well for all use-cases. Just my 2 cents.

Reply:
Oh, apologies for muddying the waters there! Take my involvement solely as a vote for the ability to group these options somehow!